oak-lucene/src/test/java/org/apache/jackrabbit/oak/plugins/index/lucene/dynamicBoost/DynamicBoostTest.java

```diff
@@ -77,101 +77,280 @@ protected void createTestIndexNode() throws Exception {
+        // minus works
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'long -titleone')]", XPATH, Arrays.asList("/test/asset2"));
+    }
+
+    // Section 2. Query dynamicboost testing
+    @Test
+    public void testQueryDynamicBoostBasic() throws Exception {
+        createAssetsIndexAndProperties(false, false);
+        prepareTestAssets();
+
+        assertEquals(
+                "[dam:Asset] as [a] /* lucene:test-index(/oak:index/test-index) :fulltext:plant " + "((jcr:content/metadata/predictedTags/plant:1 jcr:content/metadata/predictedTags/plant:1)^1.0E-4) ft:(\"plant\")\n" + "  where contains([a].[*], 'plant') */",
+                explain("//element(*, dam:Asset)[jcr:contains(., 'plant')]", XPATH));
+
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'plant')]", XPATH,
+                Arrays.asList("/test/asset1", "/test/asset2", "/test/asset3"));
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'flower')]", XPATH, Arrays.asList("/test/asset1", "/test/asset2"));
+    }
+
+    // dynamic boost: query should be case insensitive
+    @Test
+    public void testQueryDynamicBoostCaseInsensitive() throws Exception {
+        createAssetsIndexAndProperties(false, false);
+        prepareTestAssets();
+
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'FLOWER')]", XPATH, Arrays.asList("/test/asset1", "/test/asset2"));
+    }
+
+    @Ignore //todo: bug? wildcard doesn't work at all for dynamic boost term
+    @Test
+    public void testQueryDynamicBoostWildcard() throws Exception {
+        createAssetsIndexAndProperties(false, false);
+        prepareTestAssets();
+        assertQuery("//element(*, dam:Asset)[jcr:contains(., 'blu*')]", XPATH, Arrays.asList("/test/asset3"));
+    }
+
+    // dynamic boost: should respect confidence as query result order
+    @Test
+    public void testQueryDynamicBoostOrder() throws Exception {
+        createAssetsIndexAndProperties(false, false);
+        prepareTestAssets();
+
+        assertOrderedQuery("select [jcr:path] from [dam:Asset] where contains(*, 'plant')",
+                Arrays.asList("/test/asset2", "/test/asset3", "/test/asset1"));
+    }
+
+    // dynamic boost: space is explained as OR instead of AND which is documented as fulltext query in OAK doc todo: should be documented
+    @Test
+    public void testQueryDynamicBoostSpace() throws Exception {
+        createAssetsIndexAndProperties(false, false);
+        prepareTestAssets();
+
+        assertQuery("select [jcr:path] from [dam:Asset] where contains(*, 'blue flower')", SQL2,
+                Arrays.asList("/test/asset1", "/test/asset2", "/test/asset3"));
+    }
+
+    @Test
+    public void testQueryDynamicBoostOr() throws Exception {
+        createAssetsIndexAndProperties(false, false);
+        prepareTestAssets();
+
+        assertQuery("select [jcr:path] from [dam:Asset] where contains(*, 'blue OR coffee')", SQL2,
```

Comment:
I guess an explicit `AND` would not work here too. This is the expected behavior though.